proposals/explicit-backing-fields.md

```diff
@@ -14,417 +14,396 @@ but one is part of a public API and another is an implementation detail.
+which makes use of the feature completely pointless.
+
+#### Property with initializer compilation
+
+Property with explicit backing field and initializer is represented by additional
+auxiliary synthetic field which is needed for storing the value from property initializer.
+Auxiliary field is given the name `propertyName$init` and is `private` regardless of property visibility.
 
 ```kotlin
-// backing field pattern
-private var _value: Any? = UNINITIALIZED_VALUE
+val city: StateFlow<String> = field.asStateFlow()
+    field = MutableStateFlow("")
+```
 
-override val value: T
-    get() {
-       // initializes _value backing field on the first access
-    }
+The example above is compiled into:
+
+```java
+private final MutableStateFlow<String> city = MutableStateFlow("");
+private /* synthetic */ final StateFlow<String> city$init = this.city.asStateFlow();
+
+public final StateFlow<String> getCity() {
+    return this.city$init;
+}
 ```
 
-## Change log
+### Reflection
+
+Callable reference to property with explicit backing field has type `KProperty<V>` (or its subtypes) where `V`
+is type of property (not backing field) regardless of whether it is called inside class or not.
+
+On JVM backing field can be obtained using [`javaField` extension](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.reflect.jvm/java-field.html).
 
-This section records changes to this KEEP.
+There is no API in Reflection to check whether property has an explicit backing field or obtain any information about it.
+Adding such an API might be considered in [Future enhancements](#future-enhancements).
 
-### Prototype in K2 preview with Kotlin 1.7.0
+## Future enhancements
 
-Prototype of this proposal has been delivered in Kotlin 1.7.0 as a part of K2 compiler preview.
-In order to try out his new feature you need to enable the K2 compiler with 
-`-Xuse-k2` command line option and enable this language feature with `-XXLanguage:+ExplicitBackingFields`.
-The implementation is not stable yet and will generate pre-release binary. 
-There is no IDE support yet.
+1. Make it possible to access getter instead of explicit backing field when referenced within the scope of visibility of explicit backing field. This is not supported because usually there is no need for calling getter (see [Access backing property inside](#access-backing-property-inside)).
+2. Support other visibilities of explicit backing field (`protected` or `internal`).
+3. Support delegation of explicit backing field or property.
+4. Support `lateinit` explicit backing field.
+5. Support combining mutable explicit backing field and non-mutable property. 
+Despite its popularity (see [`var` backing property](#var-backing-property)), this use case is not supported in this proposal, because it is impossible for one property to behave as mutable + nullable and
+at the same time non-mutable and non-nullable.
+6. Add API in Reflection to retrieve information about explicit backing field.
```

Comment:
Since this KEEP also suggests bringing in the `init` field, it can probably raise a similar concern?